CRM/Contact/Form/Task/EmailTrait.php

```diff
@@ -702,6 +702,10 @@ public static function deprecatedTokensFormRule(array $fields) {
       '{contribution.payment_instrument}' => '{contribution.payment_instrument_id:label}',
       '{contribution.contribution_id}' => '{contribution.id}',
       '{contribution.contribution_source}' => '{contribution.source}',
+      '{contribution.contribution.status}' => '{contribution.contribution_status_id:label}',
```

Comment:
Is this right? With the "." instead of "_"? On 5.39 it appears to have been {contribution.contribution_status}

Reply:
Or no it was {contribution.contribution_status_id}. Where there two old ones maybe?

Reply:
@demeritcowboy yeah I think so - we could add ` {contribution.contribution_status_id}` to the list - the goal is to block submission of anything ambiguous

Reply:
I'm still confused a bit. Sorry this one's probably more work than you were looking for. There doesn't seem to have been a {contribution.contribution.status} before, so I'm wondering if that's a typo, and where is the conversion for {contribution.contribution_status}. i.e.

5.39 renders like
{contribution.contribution_status_id}: 1
{contribution.contribution_status}: Completed
{contribution.contribution.status}: {contribution.contribution.status}
{contribution.status}: 

This PR:
{contribution.contribution_status_id}: 1
{contribution.contribution_status}:
~~{contribution.contribution.status}: {contribution.contribution.status}~~
{contribution.status}:

Maybe I'm missing it but I don't see anything that converted {contribution.contribution_status} to {contribution.contribution_status_id:label}. It now renders as blank if you have it in your template.

Reply:
@demeritcowboy I just tested this branch locally - I selected `{contribution.contribution_status_id:label}` from the widget and it rendered to 'Completed' - the expectation is that we should be only using that token going forwards (unless you want the id or machine name)

![image](https://user-images.githubusercontent.com/336308/134726852-993cf657-2e34-4250-882c-89c72432584b.png)


![image](https://user-images.githubusercontent.com/336308/134726913-7e96afd3-0ff6-447a-92a4-12ec89a4ec92.png)

Reply:
Sorry my sentence was ambiguous. I meant {contribution.contribution_status} now renders as blank. I see you've updated the warning though now 👍 . But I don't see an upgrade task that converted {contribution.contribution_status} in message templates? Or, if it isn't fully deprecated yet, then it should still render.

Reply:
@demeritcowboy so re upgrade tasks I've been taking the following approach

- token is in scheduled reminders - upgrade civicrm_action_schedule
- token is in work flow message - upgrade the specific message
- token is in badge - upgrade civicrm_print_label
- token is in pdf or email - use form rule to block submission

I mostly didn't go with the upgrades on the last 2 because I was worried about the idea you can upload a word doc as a template & I figured as long as the advertisement is right & you can't submit the old one it would probably not be too onerous

Reply:
Ok thanks. It doesn't block this PR just if somebody had e.g. customized a workflow template to use contribution.contribution_status, that would now start coming out blank as soon as they upgrade (I didn't test that, but I think that's what would happen based on what I did test).

Reply:
@demeritcowboy now I think about it the other issue that we have been in a situation where not every place is updated to use the new tokens. However, the new token should work everywhere so I might as well put up a PR